src/main/java/github/scarsz/discordsrv/listeners/DiscordAccountLinkListener.java

```diff
@@ -45,8 +47,24 @@ public void onPrivateMessageReceived(PrivateMessageReceivedEvent event) {
 
         DiscordSRV.api.callEvent(new DiscordPrivateMessageReceivedEvent(event));
 
+        // don't link accounts if config option is disabled
+        if (!DiscordSRV.config().getBoolean("MinecraftDiscordAccountLinkedUsePM")) return;
+
+        String reply = DiscordSRV.getPlugin().getAccountLinkManager().process(event.getMessage().getContentRaw(), event.getAuthor().getId());
+        if (reply != null) event.getMessage().reply(reply).queue();
+    }
+
+    @Override
+    public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
+        // don't process messages sent by the bot
+        if (event.getAuthor().getId().equals(event.getJDA().getSelfUser().getId())) return;
+
+        // if message is not in the link channel, don't process it
+        TextChannel linkChannel = DiscordSRV.getPlugin().getDestinationTextChannelForGameChannelName("link");
+        if (!event.getChannel().equals(linkChannel)) return;
+
         String reply = DiscordSRV.getPlugin().getAccountLinkManager().process(event.getMessage().getContentRaw(), event.getAuthor().getId());
-        if (reply != null) event.getChannel().sendMessage(reply).queue();
+        if (reply != null) event.getMessage().reply(reply).queue();
```

Comment:
Would be nice to have a option to automatically delete the request and response (after 5 seconds or a different configured value) to keep the channel clean

Reply:
Ok, I implemented a feature to delete after a configured number of seconds